### ISSUE TITLE
mysql wrapper doesn't support row with NULL column.

### ISSUE THREAD

**Issue by ekarlso** [nim-lang/Nim#1837]
```
home/karlsone/projects/nim-lang/packages/test 
@[1, mysql, aaa, mit, http://vg.no, ekarlso]
@[2, amqp, 1, MIT, http://github.com/ekarlso/nim-amqp, ekarlso@gh]
Traceback (most recent call last)
test.nim(6)              test
system.nim(1979)         $
system.nim(1961)         collectionToString
SIGSEGV: Illegal storage access. (Try to compile with -d:useSysAssert -d:useGcAssert for details.)
Error: execution of an external program failed
```

**Comment by dom96** [comment]
Duplicate of #1359 and #1477

**Comment by dom96** [comment]
This is IMO still an issue and since #1477 has been closed I will reopen this. We need to figure out a way to support NULL columns in the db* modules.

**Comment by dom96** [comment]
@Araq Maybe we should change the API to use Option[T]?

**Comment by Araq** [comment]
No, let's add overloads that accept Json. It can be more efficient this way for integers etc. Option[T] would be Option[string] which is just as weird.

**Comment by dom96** [comment]
Couldn't we come up with some sort of simple type-safe API using a macro? Abusing JSON for this seems odd.

**Comment by Araq** [comment]
Hmm, ok. So it would be ``varargs[typed]``? Nice idea, could work.

**Comment by dom96** [comment]
Marking as high priority so that we do this soon.

**Comment by dom96** [comment]
This doesn't require breaking changes so I'm demoting it.

**Comment by Stuffe** [comment]
Any suggested temporary work around to this? This has got me quite stuck at the moment

**Comment by Araq** [comment]
Tried https://github.com/Senketsu/nim-db-ex ?

**Comment by Stuffe** [comment]
Araq, I ended up setting the last bit in every byte to 1 and putting those lost bits in an extra byte every 8 bytes. Insanely disgusting but avoids null bytes and I had to come up with something fast

**Comment by jfilby** [comment]
I have this problem constructing complex inserts with db_postgres. I would say the solution is to use Option[T] as it is the most straightforward and the concept of options map to null/not null quite well.
